indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,372 @@
+                <Modal.Header>
+                    {isNewArea ? (
+                        <Translate>Create new area</Translate>
+                    ) : (
+                        <Translate>Update area</Translate>
+                    )}
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="map-area-form" onSubmit={handleSubmit}>
+                        <Form.Group>
+                            <FinalInput name="name"
+                                        label={Translate.string('Name of the area')}
+                                        disabled={submitting || submitSucceeded}
+                                        autoFocus
+                                        required />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalCheckbox name="default"
+                                           label={Translate.string('Default map area')}
+                                           toggle />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalInput name="bounds.southWest.lat"
+                                        type="number"
+                                        label={Translate.string('SW Latitude')}
+                                        disabled={submitting || submitSucceeded}
+                                        validate={v.number()}
+                                        required />
+                            <FinalInput name="bounds.southWest.lng"
+                                        type="number"
+                                        label={Translate.string('SW Longitude')}
+                                        disabled={submitting || submitSucceeded}
+                                        validate={v.number()}
+                                        required />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalInput name="bounds.northEast.lat"
+                                        label={Translate.string('NE Latitude')}
+                                        type="number"
+                                        disabled={submitting || submitSucceeded}
+                                        validate={v.number()}
+                                        required />
+                            <FinalInput name="bounds.northEast.lng"
+                                        label={Translate.string('NE Longitude')}
+                                        type="number"
+                                        disabled={submitting || submitSucceeded}
+                                        validate={v.number()}
+                                        required />
+                        </Form.Group>
+                    </Form>
+                </Modal.Content>
+                <Modal.Actions>
+                    <Button type="submit" form="map-area-form"
```

Comment:
I added a FinalSubmit component when refactoring the form stuff; check if you can use that here.

Reply:
Ohh, nice. Didn't notice that one.